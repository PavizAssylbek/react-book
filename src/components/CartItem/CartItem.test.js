import React from 'react'
import CartItem from './index'

const onClick = () => {
  console.log("check function")
}



describe('should render CartItem component', () => {
  
  it('should contain id', () => {
    const component = <CartItem id={1} />
    expect(component).toMatchSnapshot()
  });

  it('should contain name and author ', () => {
    const component = <CartItem name="string" author="string" />
    expect(component).toMatchSnapshot()
  });

  it('should contain totalPrice', () => {
    const component = <CartItem totalPrice={1} />
    expect(component).toMatchSnapshot()
  });

  it('should contain totalCount', () => {
    const component = <CartItem totalCount={1} />
    expect(component).toMatchSnapshot()
  });

  it('should contain function onClick={Func}', () => {
    const component = <CartItem onRemove={onClick} onMinus={onClick} onPlus={onClick} />
    expect(component).toMatchSnapshot()
  });

})