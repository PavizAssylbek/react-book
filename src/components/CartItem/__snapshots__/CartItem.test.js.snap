// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartItem component should contain function onClick={Func} 1`] = `
<CartItem
  onClick={[Function]}
  onMinus={[Function]}
  onPlus={[Function]}
  onRemove={[Function]}
/>
`;

exports[`should render CartItem component should contain id 1`] = `
<CartItem
  id={1}
  onClick={[Function]}
/>
`;

exports[`should render CartItem component should contain name and author  1`] = `
<CartItem
  author="string"
  name="string"
  onClick={[Function]}
/>
`;

exports[`should render CartItem component should contain totalCount 1`] = `
<CartItem
  onClick={[Function]}
  totalCount={1}
/>
`;

exports[`should render CartItem component should contain totalPrice 1`] = `
<CartItem
  onClick={[Function]}
  totalPrice={1}
/>
`;
